netbox/dcim/models/devices.py

```diff
@@ -1129,3 +1136,88 @@ def delete(self, *args, **kwargs):
             )
 
         return super().delete(*args, **kwargs)
+
+
+class VirtualDeviceContext(NetBoxModel):
+    device = models.ForeignKey(
+        to='Device',
+        on_delete=models.PROTECT,
+        related_name='vdcs',
+        blank=True,
+        null=True
+    )
+    name = models.CharField(
+        max_length=64
+    )
+    status = models.CharField(
+        max_length=50,
+        blank=True,
+        choices=VirtualDeviceContextStatusChoices,
+    )
+    identifier = models.PositiveSmallIntegerField(
+        blank=True,
+        null=True,
+        validators=[MaxValueValidator(255)]
```

Comment:
Do we want to be really strict about the maximum? Just wondering if other implementations might use higher IDs.

Reply:
I suspect just sticking with PositiveSmallIntegerField and no additional validators is sufficient.